cocos/platform/android/java/src/org/cocos2dx/lib/Cocos2dxHelper.java

```diff
@@ -416,8 +416,19 @@ public static int getDPI()
                 Display d = wm.getDefaultDisplay();
                 if (d != null)
                 {
+                    try {
+                        Method getRealMetrics = d.getClass().getMethod("getRealMetrics", metrics.getClass());
+                        getRealMetrics.invoke(d, metrics);
+                        return (int)((metrics.xdpi + metrics.ydpi) / 2.0 );
+                    } catch (NoSuchMethodException e) {
+                        e.printStackTrace();
+                    } catch (IllegalAccessException e) {
+                        e.printStackTrace();
+                    } catch (InvocationTargetException e) {
+                        e.printStackTrace();
+                    }
                     d.getMetrics(metrics);
-                    return (int)(metrics.density*160.0f);
```

Comment:
According the document https://developer.android.com/reference/android/util/DisplayMetrics#density

> **This value does not exactly follow the real screen size** (as given by xdpi and ydpi), but rather is used to scale the size of the overall UI in steps based on gross changes in the display dpi. For example, a 240x320 screen will have a density of 1 even if its width is 1.8", 1.3", etc. However, if the screen resolution is increased to 320x480 but the screen size remained 1.5"x2" then the density would be increased (probably to 1.5).

We should use `xdpi & ydpi`

Reply:
Good to know. But why use `(xdpi + ydpi) / 2 `?

Reply:
For most case `xdpi & ydpi ` share the same value.  Once they are different, I
 think the average value is more practical than one of the two.

Reply:
Do you know when they will be different?

Reply:
It is depends on the display hardware.

Reply:
>  Once they are different, I think the average value is more practical than one of the two.

I am not sure. May be should use less one?

Reply:
When `xdpi` is not equals to `ydpi`, developers need to export & use `getXDPI/getYDPI`, since the DPI is not correct then.

Reply:
How do developers know the issue since it return `(xdpi + ydpi) / 2` here.

Reply:
I will add a log when the two values are different.